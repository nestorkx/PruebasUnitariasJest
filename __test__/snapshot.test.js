import { getCharacter } from "../snapshot"
import rick from '../rick'

// Cuando los datos deben ser que se envian deben ser iguales, no deben cambiar
// Guardamos un snapshot y va comparando lo guardado con lo nuevo que llega del data(API)
describe('Es hora de las instantaneas', () => {
    test('Snapshot', () => {
        // expect(getCharacter(rick)).toMatchSnapshot()
    })
    test('Siempre faltara la instantanea', () => {
        const user = {
            createAt: new Date(),
            id: Math.floor(Math.random() * 20)
        }
        // Comentamos esto para que pase la prueba
        // expect(user).toMatchSnapshot()
    })
    test('Tenemos una excepcion dentro del codigo', () => {
        const user = {
            id: Math.floor(Math.random() * 20),
            name: "Nestor Kauil"
        }
        /*expect(user).toMatchSnapshot({
            id: expect.any(Number) // El any siempre pasara la prueba ya que existe la excepcion y acepta cualquier numero
        })*/
    })
})

// jest -u
// Para actualizar el snapshot
