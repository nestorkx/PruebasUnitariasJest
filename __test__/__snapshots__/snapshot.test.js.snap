// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Es hora de las instantaneas Siempre faltara la instantanea 1`] = `
Object {
  "createAt": 2018-11-30T03:51:50.906Z,
  "id": 15,
}
`;

exports[`Es hora de las instantaneas Snapshot 1`] = `
Object {
  "gender": "Male",
  "id": 1,
  "name": "Rick Sanchez",
  "species": "Human",
  "status": "Alive",
}
`;

exports[`Es hora de las instantaneas Tenemos una excepcion dentro del codigo 1`] = `
Object {
  "id": Any<Number>,
  "name": "Nestor Kauil",
}
`;
